node/src/manager/commands/chain.rs

```diff
@@ -162,20 +169,36 @@ pub fn change_block_cache_shard(
     println!("Changing block cache shard for {} to {}", chain_name, shard);
 
     let conn = primary_store.get()?;
-    let chain = find_chain(&conn, &chain_name)?.expect("no such chain");
+
+    let chain =
+        find_chain(&conn, &chain_name)?.ok_or_else(|| anyhow!("unknown chain: {}", chain_name))?;
     let old_shard = chain.shard;
 
     println!("Current shard: {}", old_shard);
 
     let chain_store = store
         .chain_store(&chain_name)
-        .ok_or_else(|| anyhow!("unknown chain: {}", chain_name))?;
+        .ok_or_else(|| anyhow!("unknown chain: {}", &chain_name))?;
     let new_name = format!("{}-old", &chain_name);
 
     conn.transaction(|| -> Result<(), StoreError> {
         let ident = chain_store.chain_identifier.clone();
         let shard = Shard::new(shard.to_string())?;
-        // Drop the foriegn key constraint on deployment_schemas
+
+        // Fetch the current last_value from the sequence
+        let result = sql_query("SELECT last_value FROM chains_id_seq").get_result::<ChainIdSeq>(&conn)?;
```

Comment:
Wouldn't using `next_value` wouldn't that increment the sequence too than not? I manually incremented because i don't want the increment in the sequence myself but just predict the next value.